tests/utils/streams/mixins.py

```diff
@@ -148,15 +146,83 @@ def revocation_callback(partitions: Sequence[Partition]) -> None:
             message = consumer.poll(10.0)  # XXX: getting the subscription is slow
             assert isinstance(message, Message)
             assert message.partition == Partition(topic, 0)
-            assert message.offset == 1
+            assert message.offset == messages[1].offset
             assert message.payload == 1
 
             try:
                 assert consumer.poll(1.0) is None
             except EndOfPartition as error:
                 assert error.partition == Partition(topic, 0)
-                assert error.offset == 2
+                assert error.offset == message.get_next_offset()
             else:
                 raise AssertionError("expected EndOfPartition error")
 
             consumer.close()
+
+    def test_working_offsets(self) -> None:
+        with self.get_topic() as topic:
+            with closing(self.get_producer()) as producer:
+                (partition, offset) = producer.produce(topic, 0).result(timeout=5.0)
+
+            def on_assign(partitions: Mapping[Partition, int]) -> None:
+                # NOTE: This will eventually need to be controlled by a generalized
+                # consumer auto offset reset setting.
+                assert partitions == consumer.tell() == {partition: offset}
+
+            consumer = self.get_consumer("group")
+            consumer.subscribe([topic], on_assign=on_assign)
+
+            for i in range(5):
+                message: Optional[Message[int]] = consumer.poll(1.0)
+                if message is not None:
+                    break
+                else:
+                    time.sleep(1.0)
+            else:
+                raise Exception("assignment never received")
```

Comment:
This is a little clumsy — I'd like to make `subscribe` return a `Future` so that it's easier to block while waiting for an assignment, but that is going to be a more complicated endeavor because it means the consumer will need to run in a separate thread that continually polls (like the producer does) and buffers messages to be returned with the next `poll` call.

I tried to get around this by calling [`consumer.consume(0, timeout=0.0)`](https://docs.confluent.io/current/clients/confluent-kafka-python/index.html#confluent_kafka.Consumer.consume) to try and call callbacks without actually receiving messages, but it doesn't fire callbacks unless you retrieve > 1 message (because `confluent_kafka`, I guess..)